gateway-protocol/src/main/proto/gateway.proto

```diff
@@ -155,7 +155,101 @@ message CreateProcessInstanceWithResultResponse {
   // JSON document
   // consists of visible variables in the root scope
   string variables = 5;
+}
 
+message EvaluateDecisionRequest {
+  // the unique key identifying the decision to be evaluated (e.g. returned
+  // from a decision in the DeployResourceResponse message)
+  int64 decisionKey = 1;
+  // the ID of the decision to be evaluated
+  string decisionId = 2;
+  // JSON document that will instantiate the variables for the decision to be
+  // evaluated; it must be a JSON object, as variables will be mapped in a
+  // key-value fashion, e.g. { "a": 1, "b": 2 } will create two variables,
+  // named "a" and "b" respectively, with their associated values.
+  // [{ "a": 1, "b": 2 }] would not be a valid argument, as the root of the
+  // JSON document is an array and not an object.
+  string variables = 3;
+}
+
+message EvaluateDecisionResponse {
+  // the unique key identifying the decision which was evaluated (e.g. returned
+  // from a decision in the DeployResourceResponse message)
+  int64 decisionKey = 1;
+  // the ID of the decision which was evaluated
+  string decisionId = 2;
+  // the name of the decision which was evaluated
+  string decisionName = 3;
+  // the version of the decision which was evaluated
+  int32 decisionVersion = 4;
+  // the ID of the decision requirements graph where the decision which was
+  // evaluated belongs to.
+  string decisionRequirementsId = 5;
+  // the unique key identifying the decision requirements graph where the
+  // decision which was evaluated belongs to.
+  int64 decisionRequirementsKey = 6;
+  // JSON document that will instantiate the result of the decision which was
+  // evaluated; it will be a JSON object, as the result output will be mapped
+  // in a key-value fashion, e.g. { "a": 1 }.
+  string result = 7;
+  // a list of decisions that were required to be evaluated before the decision
+  // that was requested for evaluation. If empty, the decision which was
+  // evaluated didn't require any additional decision evaluations.
+  repeated EvaluatedDecision intermediateResults = 8;
```

Comment:
❌ The `DecisionEvaluationRecord` contains an entry in `getEvaluatedDecisions()` for every evaluated decision, not just the required ones. It does not just include the intermediate results but also the evaluation of the last decision. So the comment and name are not representative.

I suggest we rename this to `evaluatedDecisions` and adjust the comment. Alternatively, we filter out the evaluation of the last decision. However, I'd argue that this EvaluatedDecision contains useful information, like the `type`, `matchedRules`, `inputs, and `outputs`, that otherwise would not be available to the user.